dbrequests/mysql/tests/test_mysql_connection.py

```diff
@@ -0,0 +1,183 @@
+"""Tests against mariadb database."""
+
+import pandas as pd
+import pytest
+import numpy as np
+from dbrequests.mysql.tests.conftest import set_up_cats as reset
+from sqlalchemy.exc import InternalError
+
+
+@pytest.mark.usefixtures('db')
+class TestConnection:
+    """Unit Tests for send_data method of a mysql connection."""
+
+    def test_insert_happy_path(self, db):
+        """Insert some data and check, that it is actually there."""
+        df_add = pd.DataFrame({
+            'name': ['Chill'],
+            'owner': ['Alex'],
+            'birth': ['2018-03-03']
+        })
+
+        reset(db)
+        db.send_data(df_add, 'cats', mode='insert')
+        df_out = db.query("select name, owner, birth from cats where id = 4;")
+
+        df_out.birth = df_out.birth.astype(str)
+        assert (df_add == df_out).all(axis=None)
+
+    def test_insert_no_override(self, db):
+        """Do not override on duplicate key."""
+        df_add = pd.DataFrame({
+            'id': [3],
+            'name': ['Charlie'],
+            'owner': ['River'],
+            'birth': ['2016-05-22']
+        })
+
+        reset(db)
+        db.send_data(df_add, 'cats', mode='insert')
+        df_out = db.query("SELECT * FROM cats where id = 3;")
+
+        assert df_out.birth.astype(str).values[0] == '2016-05-21'
+
+    def test_send_data_truncate(self, db):
+        """Truncate table before insert."""
+        df_replace = pd.DataFrame({
+            'id': [1],
+            'name': ['Chill'],
+            'owner': ['Alex'],
+            'birth': ['2018-03-03']
+        })
+
+        reset(db)
+        db.send_data(df_replace, 'cats', mode='truncate')
+
+        df_nrow = db.query("SELECT count(*) as nrows FROM cats;")
+        assert df_nrow.nrows.values[0] == 1
+
+        df_out = db.query("SELECT * FROM cats;")
+        df_out.birth = df_out.birth.astype(str)
+        assert (df_replace == df_out).all(axis=None)
+
+    def test_send_data_delete(self, db):
+        """Delete before insert and check the rollback."""
+        # Testing for rollback
```

Comment:
I'd prefer having two different tests for testing two different outcomes/functionalities, for ease of debugging

Reply:
Done